src/database/schema/ChannelListSchema.ts

```diff
@@ -503,7 +503,7 @@ class ChannelList implements BaseDbSchema {
     anonUserInfo: AnonUserInfo | null = null
   ): ChannelList {
     const isPM = channelType === 'PM';
-    const firstMessage = data?.firstMessage;
+    const firstMessage = data?.messages[0] || data?.firstMessage;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change here is to assign the first message from either `data.messages[0]` or `data.firstMessage`. This could potentially introduce a bug if `data.messages` is not an array or is undefined. It would be safer to check if `data.messages` is an array before trying to access its first element.
```diff
-    const firstMessage = data?.firstMessage;
+    const firstMessage = Array.isArray(data?.messages) ? data?.messages[0] : data?.firstMessage;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->